updated with machine_config template for ignition provider

Smitha-Ram opened the pull request:
updated with machine_config template for ignition provider

TheRealDwright commented on versions.tf:
```diff
@@ -5,8 +5,17 @@ terraform {
-      source = "terraform-providers/ignition"
+    ct = {
+      source  = "poseidon/ct"
+      version = "0.7.1"
```
This provider version is not compiled for darwin arm64 (I think you need to go to a newer version)

TheRealDwright commented on versions.tf:
```diff
@@ -5,8 +5,17 @@ terraform {
+      source  = "poseidon/ct"
+      version = "0.7.1"
+    }
+    template = {
+      source  = "hashicorp/template"
+      version = "~> 2.2.0"
```
The template provider is now deprecated, see [hashicorp docs](https://registry.terraform.io/providers/hashicorp/template/latest/docs) - could you please look at using something else?

Smitha-Ram replied:
replaced template with file()

TheRealDwright commented on versions.tf:
```diff
@@ -5,8 +5,17 @@ terraform {
+      version = "~> 2.2.0"
+    }
+    null = {
+      source  = "hashicorp/null"
+      version = "~> 3.0.0"
```
Provider registry.terraform.io/hashicorp/null v3.0.0 does not have a package available for darwin_arm64

Smitha-Ram replied:
with the file(), "hashicorp/null" not required

joshdk commented on linux-config/machine.yaml:
```diff
@@ -0,0 +1,47 @@
+---
```
Having this file extracted looks so much cleaner!

joshdk replied:
I haven't tested this, but I wanted to flag that not every tool can handle multi-document yaml streams (as indicated by the `---`). Just verify that this `terraform apply`s.

joshdk commented on linux-config/machine.yaml:
```diff
@@ -0,0 +1,47 @@
+          #!/bin/bash
+          curl -sf "${aws_s3_bucket.ssh_public_keys.website_endpoint}/authorized_keys"
+
+
```
nit: extra newline

joshdk commented on ec2.tf:
```diff
@@ -16,81 +16,12 @@ data "aws_ami" "flatcar" {
-    data.ignition_file.sshd_authorized_keys.rendered,
-  ]
+locals {
+  ct_config_content = file("${path.module}/linux-config/machine.yaml")
```
This might be missing a [`templatefile()` invocation](https://www.terraform.io/language/functions/templatefile) (see above comment).

joshdk commented on ec2.tf:
```diff
@@ -16,81 +16,12 @@ data "aws_ami" "flatcar" {
-  no_create_home = true
-  shell          = "/bin/false"
+data "ct_config" "machine-ignition" {
+  content      = local.ct_config_content
```
Does this actually perform **terraform var-level** templating? The doc only say

>Validate a [Container Linux Config](https://github.com/coreos/container-linux-config-transpiler/blob/master/doc/configuration.md) or a [Butane config](https://coreos.github.io/butane/specs/) and transpile it to an [Ignition config](https://coreos.github.io/ignition/) for machine consumption. 

Which doesn't help answer that question any. 🤔

Smitha-Ram commented:
Updated machine_config template
removed terraform-ct-provide since it has some limitations with variables used in linux config template
templatefile function is used instead with var mapping

TheRealDwright reviewed (approved):
Great job, thanks for doing this!

Closes #33
